include/zephyr/device.h

```diff
@@ -950,13 +950,16 @@ BUILD_ASSERT(sizeof(device_handle_t) == 2, "fix the linker scripts");
 	Z_INIT_ENTRY_DEFINE(DEVICE_NAME_GET(dev_name), init_fn,		\
 		(&DEVICE_NAME_GET(dev_name)), level, prio)
 
+/* device_extern is generated based on devicetree nodes */
+#undef DEVICE_ORD_MACRO
+#define DEVICE_ORD_MACRO(x) extern const struct device DEVICE_DT_NAME_GET(x)
+
+#include <device_ord.h>
```

Comment:
why do we need undef here? shouldn't `DEVICE_ORD_MACRO` be in `device_ord.h`? I guess it's related to topics discussed in dev-review (sorry I couldn't join), could other usages  be detailed?